server/controllers/kaban.js

```diff
@@ -0,0 +1,287 @@
+const { deleteColumnTask } = require("../utils/deleteColumnTask");
+const { generateBoard } = require("../utils/generateBoards");
+const { updateBoardBoard } = require("../utils/updateBoardBoard");
+const { updateBoardColumn } = require("../utils/updateBoardColumn");
+const { updateColumnTask } = require("../utils/updateColumnTask");
+
+// @route GET /kaban/homeBoard
+// @access Protected
+exports.homeBoard = asyncHandler(async (req, res,) => {
+ try {
+  const userId = req.user.id;
+  let boards = await generateBoard(userId);
+  res.status(200)
+  res.json({ success: true, boards })
+ } catch (error) {
+  res.status(501)
+  throw new Error("Error Processing your request at this time")
+ }
+});
+
+// @route POST /kaban/board
+// @access Protected
+exports.createBoard = asyncHandler(async (req, res,) => {
+ try {
+  const userId = req.user.id;
+  const { name } = req.body;
+  let board = await createBoardBoard({ name, userId });
+  res.status(200)
+  res.json({ success: true, board })
+ } catch (error) {
+  res.status(501)
+  throw new Error("Error Processing your request at this time")
+ }
+});
+
+// @route PUT /kaban/board
+// @access Protected
+exports.updateBoard = asyncHandler(async (req, res,) => {
+ try {
+  const userId = req.user.id;
+  const { name } = req.body;
+  const { board_id } = req.params;
+  let board = await Board.findOne({ _id: board_id });
+  if (!board) {
+   res.status(401)
+   return res.send("Column's Board does not exist")
+  }
+  if (board.userId != userId) {
+   res.status(401)
+   return res.send("Hey!😁, you are not authorized")
+  }
+  let updatedBoard = await updateBoardBoard({ name }, board_id)
+  res.status(200)
+  res.json({ success: true, board: updatedBoard })
+ } catch (error) {
+  res.status(501)
+  throw new Error("Error Processing your request at this time")
+ }
+});
+
+// @route DELETE /kaban/board
+// @access Protected
+exports.deleteBoard = asyncHandler(async (req, res,) => {
+ try {
+  const userId = req.user.id;
+  const { board_id } = req.params;
+  let board = await Board.findOne({ _id: board_id });
+  if (!board) {
+   res.status(401)
+   return res.send("Column's Board does not exist")
+  }
+  if (board.userId != userId) {
+   res.status(401)
+   return res.send("Hey!😁, you are not authorized")
+  }
+  await deleteBoardBoard(board_id);
+  res.status(200)
+  res.json({ success: true })
+ } catch (error) {
+  res.status(501)
+  throw new Error("Error Processing your request at this time")
+ }
+});
+
+// @route POST /column/:board_id
+// @access Protected
+exports.createColumn = asyncHandler(async (req, res,) => {
+ try {
+  const userId = req.user.id;
+  const { name, index } = req.body;
+  const { board_id } = req.params;
+  let board = await Board.findOne({ _id: board_id });
+  if (!board) {
+   res.status(401)
+   return res.send("Column's Board does not exist")
+  }
+  if (board.userId != userId) {
+   res.status(401)
+   return res.send("Hey!😁, you are not authorized")
```

Comment:
Let's chain these together. No need for a `return` statement as `res.send` already terminates request/response cycle.